kotlin/internal/toolchains.bzl

```diff
@@ -202,6 +203,15 @@ _kt_toolchain = rule(
             `kt_abi_plugin_incompatible`""",
             default = False,
         ),
+        "experimental_use_public_only_abi_jars": attr.bool(
+            doc = """Compile using public only abi jars.
+            This effectively applies the following two compiler plugin options.
+            plugin:org.jetbrains.kotlin.jvm.abi:treatInternalAsPrivate=true
+            plugin:org.jetbrains.kotlin.jvm.abi:removePrivateClasses=true
+            Can be disabled for an individual target using the tag.
+            `kt_abi_plugin_incompatible`""",
```

Comment:
This should probably have a tag of its own to disable this behavior at the target level so that adoption across larger projects is easier.